onboard/catkin_ws/src/task_planning/scripts/combination_tasks.py

```diff
@@ -0,0 +1,125 @@
+
+from task import Task
+
+
+class ListTask(Task):
+    """Run a list of tasks sequentially"""
+
+    def __init__(self, tasks, num_loops=1):
+        super(ListTask, self).__init__()
+
+        self.num_loops = num_loops
+        self.tasks = tasks
+        self.curr_index = 0
+
+    def _on_task_run(self):
+        if self.curr_index == len(self.tasks):
+            if self.num_loops > 0:
+                self.num_loops -= 1
+                if self.num_loops == 0:
+                    self.finish()
+
+            self.curr_index = 0
+
+        elif self.tasks[self.curr_index].finished:
+            if self.num_loops != 1:
+                self.tasks[self.curr_index].restart()
+            self.curr_index += 1
+
+        else:
+            self.tasks[self.curr_index].run()
+
+
+class IndSimulTask(Task):
```

Comment:
I think you can combine IndSimulTask and DepSimulTask into one, they use basically the same code, only opposite bool variables based on their task which can be set as an input to the constructor.

Reply:
We'll re-evaluate these with the rewrite.